guides/migrate-from-v2.md

```diff
@@ -0,0 +1,259 @@
+  we try to be strict in this differentiation. Also layers aren't divided into
+  baselayers and overlays any more.
+* `ol.interaction.<Whatever>` Interactions react to user input and change the
+  state of the map. They do not render DOM elements, some may render features on
+  the map.
+* `ol.control.<Whatever>` A control is a visible widget with a DOM element in a
+  fixed position on the screen. They can involve user input (buttons), or be
+  informational only. Their position is determined using CSS. What were controls
+  in OpenLayers 2 can be an interaction or a control in OpenLayers v3.
+
+## Consistency
+
+In OpenLayers 2 certain properties would have getters, e.g. to get the zoomlevel
+of a map one could use the following code:
+
+```javascript
+var map = new Openlayers.Map(/* config */);
+var zoom = map.getZoom();
+// sadly, map.setZoom does not exist
+```
+
+The appropriate setter for the `zoom` property was missing though. In practice
+people would then use other method for setting the zoom (e.g. `setCenter`) or
+manipulate the property on the instance directly (e.g. `map.zoom = 5;`).
+
+OpenLayers 3 tries to be more consistent here. Usually (most cases) you will
+find a setter for a property where we also have a getter:
+
+```javascript
+var map = new ol.Map( /* config */ );
+var zoom = map.getView().getZoom();
+// map.getView().setZoom()
+```
+
+If there is no such setter, we consider the property to be read-only.
+
+Most classes in OpenLayers 3 also inherit from the ol.Object class, which gives
+them a generic getter and setter for arbitrary properties. These can also be
+monitored for change events:
+
+```javascript
+var map = new ol.Map( /* config */ );
+map.on('change:humpty', function()({ /* handle the new value for humpty */ });
+map.set('humpty', 123);
+map.get('humpty');
+map.set('humpty', 456);
+```
+
+## Examples
+
+Let's have a look at some of the examples of the version 2 and see how we can
+write these as Openlayers 3 examples.
+
+### WMS layer
+
+The code behind the [OpenLayers 2 WMS
+example](http://dev.openlayers.org/examples/wms.html) mainly is:
+
+```javascript
+var lon = 5;
+var lat = 40;
+var zoom = 5;
+var map, layer;
+
+function init(){
+    map = new OpenLayers.Map( 'map' );
+    layer = new OpenLayers.Layer.WMS( "OpenLayers WMS",
+            "http://vmap0.tiles.osgeo.org/wms/vmap0", {layers: 'basic'} );
+    map.addLayer(layer);
+
+    map.setCenter(new OpenLayers.LonLat(lon, lat), zoom);
+}
+```
+
+Rewritten against the new API this becomes:
+
+```javascript
+var lon = 5;
+var lat = 40;
+var zoom = 5;
+var map, layer;
+
+function init(){
+  layer = new ol.layer.Tile({
+    source: new ol.source.TileWMS({
+      url: 'http://vmap0.tiles.osgeo.org/wms/vmap0',
+      params: { LAYERS: 'basic' }
+    })
+  });
+
+  map = new ol.Map({
+    layers: [layer],
+    target: 'map',
+    view: new ol.View({
+      center: [lon, lat],
+      zoom: zoom,
+      projection: 'EPSG:4326'
```

Comment:
should we say anything here that the default projection from OpenLayers 2 differs from that of ol3?